lib/portlayer/exec/exec.go

```diff
@@ -113,11 +113,14 @@ func Init(ctx context.Context, sess *session.Session, source extraconfig.DataSou
 func eventCallback(ie events.Event) {
 	// grab the container from the cache
 	container := Containers.Container(ie.Reference())
+
 	if container != nil {
+		container.m.Lock()
```

Comment:
Yes this will be blocked momentarily until the Commit exits.

However, other calls like NewHandle will also be blocked by the lock in Commit; which is what @caglar10ur was seeing originally.